### DISCUSSION TITLE
Problem with USB CDC and STM32F105

### DISCUSSION THREAD

**Discussion by Ereaey** [zephyrproject-rtos/zephyr#60097] [Q&A]
Hello,

After several attempts and extensive research, I am unable to see my USB in dmesg. I'm using an external quartz with an up-to-date dts configuration. I have an i2c and a uart that work correctly.

Here is my configuration:

```c
&clk_lsi {
    status = "okay";
};

&clk_hse {
    clock-frequency = <DT_FREQ_K(14745)>;
    status = "okay";
};

&pll2 {
    clocks = <&clk_hse>;
    mul = <9>;
    prediv = <3>;
    status = "okay";
};

&pll {
    mul = <15>; //Normally 6.5 but needs to be 15 to be used
    prediv = <4>;
    clocks = <&pll2>;
    status = "okay";
};

&rcc {
    clocks = <&pll>;
    clock-frequency = <DT_FREQ_K(71881)>;
    ahb-prescaler = <1>;
    apb1-prescaler = <2>;
    apb2-prescaler = <1>;
};

zephyr_udc0: &usbotg_fs {
    pinctrl-0 = <&usb_otg_fs_dm_pa11 &usb_otg_fs_dp_pa12>;
    pinctrl-names = "default";
    status = "okay";
    cdc_acm_uart0 {
        compatible = "zephyr,cdc-acm-uart";
    };
};
```

The prj contains:

```c
CONFIG_STDOUT_CONSOLE=y
CONFIG_USB_DEVICE_STACK=y
CONFIG_USB_DEVICE_PRODUCT="Zephyr CDC ACM sample"
CONFIG_USB_DEVICE_PID=0x0002
CONFIG_LOG=y
CONFIG_USB_DRIVER_LOG_LEVEL_ERR=y
CONFIG_USB_DEVICE_LOG_LEVEL_ERR=y
CONFIG_SERIAL=y
CONFIG_UART_INTERRUPT_DRIVEN=y
CONFIG_UART_LINE_CTRL=y
CONFIG_USB_DEVICE_INITIALIZE_AT_BOOT=n
```

And only a basic USB initialization:

```c
const struct device *dev;
uint32_t baudrate, dtr = 0U;
int ret;

dev = DEVICE_DT_GET_ONE(zephyr_cdc_acm_uart);
if (!device_is_ready(dev)) {
    return false;
}

ret = usb_enable(NULL);

if (ret != 0) {
    return false;
}
return true;
```

I do not get any errors, I am at my wit's end, I have checked the code and my card but nothing works. Any ideas?

Thank you!

**Comment by FRASTM** [comment]
I see 2 points that could be checked 
-  With the value set for the clock scheme, the USB clock is seen at 47.92MHz, I guess this is valid (compared to 48MHz)
-  the DTS of the stm32f105.dtsi configures the `num-bidir-endpoints = <4>;`  but the refman [RM0008](https://www.st.com/resource/en/reference_manual/rm0008-stm32f101xx-stm32f102xx-stm32f103xx-stm32f105xx-and-stm32f107xx-advanced-armbased-32bit-mcus-stmicroelectronics.pdf), mentions 3+3 bidir end points
